### PR TITLE
Add new todo dashboard

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
This PR adds the new todo dashboard to the dashboard. It takes the place of the HIX analysis box. The HIX analysis box is instead moved to the translation coverage section.

### Proposed changes

- Add new todo dashboard box
- Add template for todos
- Add six todos for the first version of the todo dashboard
- Move the HIX analysis to translation coverage

### Side effects

I think none

### Resolved issues

Fixes: #2107

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -4646,10 +4646,144 @@ msgstr "Seite neu laden"
 msgid "Continue reading"
 msgstr "Weiterlesen"
 
+#: cms/templates/dashboard/_todo_dashboard_widget.html
+msgid "To-Do's"
+msgstr "To-Do’s"
```
Comment (timoludwig) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:4651]:
```suggestion
msgid "To-Do's"
msgstr "To-Dos"
```
To be fair, this was already incorrect in the design :sweat_smile:
https://www.duden.de/rechtschreibung/To_do
http://www.deppenapostroph.info/

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 2]
```diff
@@ -42,51 +46,176 @@ def get_context_data(self, **kwargs):
...
+                    last_updated__lte=OUTDATED_THRESHOLD_DATE.date(),
+                    page__explicitly_archived=False,
+                    language__slug=self.request.region.default_language.slug)
+                .order_by("page__id", "language__id", "-version", "last_updated")
+                .distinct("page__id", "language__id")
+                .all()
+        )
+        print(OUTDATED_THRESHOLD_DATE, outdated_pages[0].last_updated)
+        for p in outdated_pages:
+            # TODO: why does last_updated sometimes differ??
```
Comment (JoeyStk) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:156]: @david-venhoff This is the line we spoke about on Tuesday. If you checkout this branch and go to the dashboard view, you'll see in brackets the amount of pages that fit the criteria of being outdated (being older than 365 days). If you filter for the same criteria in the page tree (or click on the title link "outdated page") you might notice, that the amount of pages differ. In my case for 1088 pages the dashboard shows 24 outdated pages in the dashboard, but only 13 in the filtered page tree view

File: integreat_cms/cms/templates/dashboard/_todo_dashboard_widget.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <div class="flex flex-wrap gap-2">
+        <p>
+            {% blocktranslate trimmed %}
+                Every day you can find a list of ideas on how you can improve the content of your pages
+            {% endblocktranslate %}
+        </p>
+        {% include "dashboard/todo_dashboard_rows/_unreviewed_pages_row.html" %}
+        {% include "dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html" %}
+        {% include "dashboard/todo_dashboard_rows/_unread_feedback_row.html" %}
+        {% include "dashboard/todo_dashboard_rows/_broken_links_row.html" %}
```
[the remark this answers is not in the document]
Comment (JoeyStk) [reply integreat_cms/cms/templates/dashboard/_todo_dashboard_widget.html:19]: After a chat with @osmers it was decided to remove this section

File: integreat_cms/cms/templates/dashboard/_todo_dashboard_row.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <div>
+        <a href="{% spaceless %}{% block todo_dashboard_title_link %}{% endblock todo_dashboard_title_link %}{% endspaceless %}"
+           class="text-blue-500 font-bold underline">
+            {% block todo_dashboard_title %}
+            {% endblock todo_dashboard_title %}
+        </a>
+        <span>&nbsp;(in total
+            {% block todo_dashboard_number %}
+            {% endblock todo_dashboard_number %}
+        )</span>
```
[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/templates/dashboard/_todo_dashboard_row.html]: This is correct, unfortunately the html spec has a few limitations here. But in my opinion, the UI that is visible to the user is more important than code style. Improvements to the code style are only desirable if they don't influence the layout.

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 5]
```diff
@@ -42,51 +47,189 @@ def get_context_data(self, **kwargs):
...
+        r"""
+        Extend context by info on automatically saved pages
+
+        :return: Dictionary containing the context for auto saved pages
+        :rtype: dict
+        """
+        automatically_saved_pages = (
+            PageTranslation.objects.filter(
+                status=status.AUTO_SAVE,
+                page__explicitly_archived=False,
```
Comment (david-venhoff) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:219]: It would probably also make sense to exclude implicitely archived pages here.

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 6]
```diff
@@ -42,51 +47,189 @@ def get_context_data(self, **kwargs):
...
+            PageTranslation.objects.filter(
+                status=status.REVIEW,
+                page__explicitly_archived=False,
+                language__slug=self.request.region.default_language.slug,
+                page__id__in=Subquery(not_implicitly_archived_ids.values("pk")),
+            )
+            .order_by("page__id", "language__id", "-version")
+            .distinct("page__id", "language__id")
+            .all()
         )
```
[the remark this answers is not in the document]
Comment (david-venhoff) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:94]: I mixed up the draft and review statuses

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 7]
```diff
@@ -42,51 +47,189 @@ def get_context_data(self, **kwargs):
...
+        :rtype: dict
+        """
+        automatically_saved_pages = (
+            PageTranslation.objects.filter(
+                status=status.AUTO_SAVE,
+                page__explicitly_archived=False,
+                language__slug=self.request.region.default_language.slug,
+            )
+            .order_by("page__id", "language__id", "-version")
+            .distinct("page__id", "language__id")
```
[the remark this answers is not in the document]
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:226]: I think this was required, but I don't remember how exactly it went wrong without. Anyway, this should not be a performance impact even if it should not be required, so we're being lazy and will just leave it in :upside_down_face:

[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:226]: If you order by `language` instead of `language__id`, it will take the default ordering of the language model (the `bcp47_tag`):
https://github.com/digitalfabrik/integreat-cms/blob/fc529eaf8ccecbe21cfdff267c20dec07fbdfb70/integreat_cms/cms/models/languages/language.py#L215
Which means it needs to query the foreign object in order to perform the query. If we order by the language's `id` however, Django doesn't need to query this additional information from the db since the language id is already stored in the translation model.

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 8]
```diff
@@ -42,51 +47,189 @@ def get_context_data(self, **kwargs):
...
+        :rtype: dict
+        """
+        OUTDATED_THRESHOLD_DATE = datetime.now() - relativedelta(
+            days=settings.OUTDATED_THRESHOLD_DAYS
+        )
+
+        outdated_pages = (
+            PageTranslation.objects.filter(
+                last_updated__lte=OUTDATED_THRESHOLD_DATE.date(),
+                page__explicitly_archived=False,
```
[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:167]: @PeterNerlich Why did you resolve this conversation? I don't see how **im**plicitly archived pages are excluded in this query...

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html [hunk 9]
```diff
@@ -0,0 +1,36 @@
+{% extends "../_todo_dashboard_row.html" %}
+{% load i18n %}
+{% block todo_dashboard_icon %}
+    save
+{% endblock todo_dashboard_icon %}
+{% block todo_dashboard_title_link %}
+    {% url 'pages' region_slug=request.region.slug language_slug=default_language_slug %}
```
[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html:7]: > Another (probably not very important) problem is that there seems to be no way to get rid of an autosave that should not get published.

@david-venhoff what do you mean by this? You can reject unwanted autosaves in the version view, right? Then it will revert to the latest public version which should result in the autosave no longer being present on the dashboard.

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    {% else %}
+        {% blocktranslate trimmed %}
+            At the moment there is no page waiting for a review. Good job!
+        {% endblocktranslate %}
+    {% endif %}
+{% endblock todo_dashboard_description %}
+{% block todo_dashboard_button_link %}
+    {% if unreviewed_pages %}
+        <a class="btn !rounded-full"
+           href="{% url 'edit_page' region_slug=request.region.slug language_slug=default_language_slug page_id=single_unreviewed_page.page_id %}">
```
Comment (timobrembeck) [inline integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html:30]:
```suggestion
           href="{% url 'edit_page' region_slug=request.region.slug language_slug=default_language_slug page_id=unreviewed_pages.0.page_id %}">
```
No need to define a new variable just to access the first element

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    {% translate "Pages that are waiting for review" %}
+{% endblock todo_dashboard_title %}
+{# djlint:off #}
+{% block todo_dashboard_number %}
+    {{ unreviewed_pages|length }}{% endblock todo_dashboard_number %}
+{# djlint:on #}
+{% block todo_dashboard_description %}
+    {% if unreviewed_pages %}
+        {% blocktranslate trimmed %}
+            The page <b>{{ single_unreviewed_page }}</b> is waiting for a review. Please review, if the content on the page is correct.
```
Comment (timobrembeck) [inline integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html:19]:
```suggestion
        {% blocktranslate trimmed with single_unreviewed_page=unreviewed_pages.0 %}
            The page <b>{{ single_unreviewed_page }}</b> is waiting for a review. Please review, if the content on the page is correct.
```

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 12]
```diff
@@ -42,51 +47,192 @@ def get_context_data(self, **kwargs):
...
+            .distinct("page__id", "language__id")
+            .all()
         )
 
-        # Get the latest versions of the page translations for these pages
-        hix_translations = PageTranslation.objects.filter(
-            language__slug__in=settings.TEXTLAB_API_LANGUAGES, page__in=hix_pages
-        ).distinct("page_id", "language_id")
+        single_unreviewed_page = unreviewed_pages[0] if unreviewed_pages else None
+
```
Comment (timobrembeck) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:97]:
```suggestion
```

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+{% block todo_dashboard_title %}
+    {% translate "Automatically saved pages" %}
+{% endblock todo_dashboard_title %}
+{# djlint:off #}
+{% block todo_dashboard_number %}
+    {{ automatically_saved_pages|length }}{% endblock todo_dashboard_number %}
+{# djlint:on #}
+{% block todo_dashboard_description %}
+    {% if automatically_saved_pages %}
+        {% blocktranslate trimmed %}
```
Comment (timobrembeck) [inline integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html:18]:
```suggestion
        {% blocktranslate trimmed with single_automatically_saved_page=automatically_saved_pages.0 %}
```

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    {% else %}
+        {% blocktranslate trimmed %}
+            At the moment there are no automatically saved pages. Good job!
+        {% endblocktranslate %}
+    {% endif %}
+{% endblock todo_dashboard_description %}
+{% block todo_dashboard_button_link %}
+    {% if automatically_saved_pages %}
+        <a class="btn !rounded-full"
+           href="{% url 'edit_page' region_slug=request.region.slug language_slug=default_language_slug page_id=single_automatically_saved_page.page_id %}">
```
Comment (timobrembeck) [inline integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html:30]:
```suggestion
           href="{% url 'edit_page' region_slug=request.region.slug language_slug=default_language_slug page_id=automatically_saved_pages.0.page_id %}">
```

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 15]
```diff
@@ -42,51 +47,192 @@ def get_context_data(self, **kwargs):
...
+            .all()
         )
 
-        # Get the number of translations which are not ready for MT
-        not_ready_for_mt_count = sum(
-            pt.hix_score < settings.HIX_REQUIRED_FOR_MT
-            for pt in hix_translations_with_score
+        single_automatically_saved_page = (
+            automatically_saved_pages[0] if automatically_saved_pages else None
         )
```
Comment (timobrembeck) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:232]:
```suggestion
```

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 16]
```diff
@@ -42,51 +47,192 @@ def get_context_data(self, **kwargs):
...
 
         return {
-            "worst_hix_translations": worst_hix_translations,
-            "hix_threshold": settings.HIX_REQUIRED_FOR_MT,
-            "ready_for_mt_count": len(hix_translations) - not_ready_for_mt_count,
-            "total_count": len(hix_translations),
+            "automatically_saved_pages": automatically_saved_pages,
+            "single_automatically_saved_page": single_automatically_saved_page,
```
Comment (timobrembeck) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:236]:
```suggestion
```

File: integreat_cms/core/settings.py [hunk 17]
```diff
@@ -1093,3 +1093,10 @@
 
 # Our database operations should never exceed 60 seconds
 DB_MUTEX_TTL_SECONDS = 60
+
+#############
+# Dashboard #
+#############
+
+# Days since a page is considered to be outdated in the todo dashboard
```
Comment (timobrembeck) [inline integreat_cms/core/settings.py:1101]:
```suggestion
#: Days after which a page is considered to be outdated in the todo dashboard
```

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 18]
```diff
@@ -42,51 +47,192 @@ def get_context_data(self, **kwargs):
...
+
+        :return: Dictionary containing the context for broken links
+        :rtype: dict
+        """
+        invalid_urls = filter_urls(self.request.region.slug, "invalid")[0]
+        invalid_url = invalid_urls[0] if invalid_urls else None
+
+        relevant_translation = (
+            invalid_url.region_links[0].content_object if invalid_url else None
+        )
```
Comment (timobrembeck) [inline integreat_cms/cms/views/dashboard/dashboard_view.py:203]: This cannot work, because `Url` objects are different from `Link` objects. The `Url` only contains the reference to the `Link` objects, and only the `Link` objects references the translation objects.

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_broken_links_row.html [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+    {% else %}
+        {% blocktranslate trimmed %}
+            At the moment there are no broken links. Good job!
+        {% endblocktranslate %}
+    {% endif %}
+{% endblock todo_dashboard_description %}
+{% block todo_dashboard_button_link %}
+    {% if broken_links %}
+        <a class="btn !rounded-full"
+           href="{% url 'edit_url' region_slug=request.region.slug url_filter='invalid' url_id=relevant_url.page_id %}#replace-url">
```
Comment (timobrembeck) [inline integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_broken_links_row.html:30]: This cannot work, the `Url` model does not have a `page_id` property or field.

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 5 above)
Comment (timobrembeck) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:222]: @PeterNerlich Here as well, you resolved the conversation, but the issue does not seem to be resolved?

File: integreat_cms/cms/templates/dashboard/_todo_dashboard_row.html [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+            <a href="{% spaceless %}{% block todo_dashboard_title_link %}{% endblock todo_dashboard_title_link %}{% endspaceless %}"
+               class="text-blue-500 font-bold underline">
+                {% block todo_dashboard_title %}
+                {% endblock todo_dashboard_title %}
+            </a>
+            {# djlint:off #}
+            <span>({% translate "in total" %}
+                {% block todo_dashboard_number %}
+                {% endblock todo_dashboard_number %})</span>
+            {# djlint:on #}
```
[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/templates/dashboard/_todo_dashboard_row.html]: @JoeyStk this is not yet resolved, you only implemented the suggested changes of the super template, but not in the sub templates. This means that you completely override the block content, because you didn't include the `{{ block.super }}` parts in the sub templates.

**Review by timobrembeck — CHANGES_REQUESTED** [review]

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html (hunk 9 above)
[the remark this answers is not in the document]
Comment (david-venhoff) [reply integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_automatically_saved_pages_row.html:7]: Ah, I was not aware of that. Then you can ignore this point ^^

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:167]: @timobrembeck Because Johannes and I went over this today. The changes are just not pushed yet. We will continue work on this tomorrow morning.

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 5 above)
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:222]: Because Johannes and I went over this today. The changes are just not pushed yet. We will continue work on this tomorrow morning.

File: integreat_cms/cms/views/dashboard/dashboard_view.py [hunk 21]
```diff
@@ -42,51 +47,192 @@ def get_context_data(self, **kwargs):
...
+
+        not_implicitly_archived_ids = Page.objects.filter(
+            id=Case(
+                When(
+                    Exists(explicitly_archived_ancestors),
+                    then=None,
+                ),
+                default=F("pk"),
+            )
+        )
```
[the remark this answers is not in the document]
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py:82]: @timobrembeck I played around with this for a bit and the tests don't seem to break overall. Except for the PDF tests because they always make issues, and `test_sitemap` because the query count went down by one:

image

Should I add this as a separate PR rather than as part of this one? If so, this one should depend on it and only get merged after it, correct?

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html (hunk 11 above)
Comment (PeterNerlich) [reply integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html]: Do you have any preference of 
```jinja
        {% blocktranslate trimmed with single_unreviewed_page=unreviewed_pages.0 %}
            The page <b>{{ single_unreviewed_page }}</b> is waiting for a review. Please review, if the content on the page is correct.
```
over
```jinja
        {% blocktranslate trimmed %}
            The page <b>{{ unreviewed_pages.0 }}</b> is waiting for a review. Please review, if the content on the page is correct.
```
or is that just because that variable was used earlier?

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 18 above)
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py]: But isn't that exactly what the line for `relevant_translation` does? It picks the first region link for an URL and selects its `content_object`

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 21 above)
[the remark this answers is not in the document]
Comment (PeterNerlich) [reply integreat_cms/cms/views/dashboard/dashboard_view.py]: I committed my suggestion directly to this branch for now.

File: integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html (hunk 11 above)
Comment (timobrembeck) [reply integreat_cms/cms/templates/dashboard/todo_dashboard_rows/_unreviewed_pages_row.html]: Exactly, I think in general it's better to avoid the definition of new variable names for trivial stuff, but in the translation files it's preferable to have more speaking variable names to get the context for the translation.

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 18 above)
Comment (timobrembeck) [reply integreat_cms/cms/views/dashboard/dashboard_view.py]: Ah damn, I'm sorry, I guess I missed the `region_links[0]` part somehow :see_no_evil:

File: integreat_cms/cms/models/pages/page.py [hunk 22]
```diff
@@ -216,14 +216,11 @@ def explicitly_archived_ancestors(self):
...
+        :return: The list of archived ancestors
         :rtype: list [ ~integreat_cms.cms.models.pages.page.Page ]
         """
-        return [
-            ancestor
-            for ancestor in self.get_cached_ancestors()
-            if ancestor.explicitly_archived
-        ]
+
+        return self.region.explicitly_archived_ancestors_query(fixed=self)
```
Comment (timobrembeck) [inline integreat_cms/cms/models/pages/page.py:223]:
```suggestion
        return [
            ancestor
            for ancestor in self.get_cached_ancestors()
            if ancestor.explicitly_archived
        ]
```
Hmm, I'd like to revert this change since it bypasses the cached ancestors and might cause additional queries in some circumstances.

File: integreat_cms/cms/models/regions/region.py [hunk 23]
```diff
@@ -630,32 +630,73 @@ def get_language_or_404(self, language_slug, only_active=False, only_visible=Fal
...
+            rgt = fixed.rgt
+        else:
+            tree = models.OuterRef("tree_id")
+            lft = models.OuterRef("lft")
+            rgt = models.OuterRef("rgt")
+
+        return self.pages.filter(
+            tree_id=tree,
+            lft__lt=lft,
+            rgt__gt=rgt,
```
Comment (timobrembeck) [inline integreat_cms/cms/models/regions/region.py:672]:
```suggestion
        return self.pages.filter(
            tree_id=models.OuterRef("tree_id"),
            lft__lt=models.OuterRef("lft"),
            rgt__gt=models.OuterRef("rgt"),
```
I think one method for the subquery is enough here.

File: integreat_cms/cms/models/regions/region.py [hunk 24]
```diff
@@ -694,16 +736,15 @@ def get_pages(
...
-        To perform the extra effort of returning an unrestricted queryset, set the parameter
-        ``return_unrestricted_queryset`` to ``True``.
-
         :param archived: Whether or not only archived pages should be returned (default: ``False``)
         :type archived: bool
 
         :param return_unrestricted_queryset: Whether or not the result should be returned as unrestricted queryset.
                                              (default: ``False``)
+
+                                             .. deprecated:: 2023.10.0
```
Comment (timobrembeck) [inline integreat_cms/cms/models/regions/region.py:745]: Hmm, I don't think we need deprecation grace periods for our functions, just removing them without notice is fine I guess.
Also, please remove all usages of the `return_unrestricted_queryset` parameter since it's not needed anymore.

File: integreat_cms/cms/views/dashboard/dashboard_view.py (hunk 21 above)
[the remark this answers is not in the document]
Comment (timobrembeck) [reply integreat_cms/cms/views/dashboard/dashboard_view.py]: @PeterNerlich wow, that's really cool, awesome improvement! :muscle:
It's fine to include this change in this PR, however could you clean up the git history a bit to have one commit for the dashboard changes and one commit for the refactoring of the model methods?

**Review by timobrembeck — CHANGES_REQUESTED** [review]
Thanks a lot! :muscle: 
The performance problems definitely have been resolved! :tada:
I'd just change a small implementation detail :wink:

**Review by timobrembeck — CHANGES_REQUESTED** [review]
Great, thanks! :+1:

@PeterNerlich there is still one occurrence of the outdated argument:

https://github.com/digitalfabrik/integreat-cms/blob/0c742982837e51281b5a86a5d9c3b04b38876036/integreat_cms/cms/views/regions/region_update_view.py#L67

This will cause an error when archiving regions via the form. It's unfortunate that we don't have a test case for this, maybe we should add one? We could add a copy of this test case:

https://github.com/digitalfabrik/integreat-cms/blob/0c742982837e51281b5a86a5d9c3b04b38876036/tests/cms/views/view_config.py#L464-L479

just with `ARCHIVED` instead of `ACTIVE`, this should then correctly trigger the `if` condition and reveal the error.

**Comment by PeterNerlich** [comment]
Memo to ourselves: Notify marketing team in time before this is released

**Comment by timobrembeck** [comment]
@PeterNerlich Then we should probably notify the team now? I think this PR is ready to be merged soon and will probably be released next week?

**Comment by JoeyStk** [comment]
@timobrembeck  & @PeterNerlich 
> @PeterNerlich Then we should probably notify the team now? I think this PR is ready to be merged soon and will probably be released next week?

I finally finished and uploaded the blog post for the to do board and it's currently being spell-checked and proof read by the marketing team. As soon as this is finished, we can release this PR :smiley: 

However I still have one question for @timobrembeck. At the conference we discussed that it's always desirable to load the dashboard via AJAX to make sure that in no instance the page doesn't load due to this widget and instead only this singluar widget is not being loaded (but navigation etc. are being loaded). What do you think? Do you agree and we should still implement the AJAX call for this widget, and if so, should we still do it before the release (and maybe even in this PR) or can we postpone this, since currently there is no bulk of database calls?

**Comment by timobrembeck** [comment]
> However I still have one question for @timobrembeck. At the conference we discussed that it's always desirable to load the dashboard via AJAX to make sure that in no instance the page doesn't load due to this widget and instead only this singular widget is not being loaded (but navigation etc. are being loaded). What do you think? Do you agree and we should still implement the AJAX call for this widget, and if so, should we still do it before the release (and maybe even in this PR) or can we postpone this, since currently there is no bulk of database calls?

Thanks to @PeterNerlich's optimizations I think AJAX-calls are optional here, I did not experience longer page loading times with this PR.
After merging this to develop, we can do a bit more testing with (nearly) live data in the test cms, and if no issues arise, I think it's fine to release it in this state and implement the ajax calls later.

**Comment by PeterNerlich** [comment]
Reminded me that I didn't fix the last bit @timobrembeck  mentioned yet. Done now!

**Review by timobrembeck — APPROVED** [review]
Thanks a lot! :+1:
I don't have time for a detailed review at the moment, but I don't see major blockers for this PR. Let's just merge it :shrug:

**Review by david-venhoff — APPROVED** [review]
I just did a quick review, but so far everything looks good 🎉
